parallel-install/pkg/jobmanager/README.md

```diff
@@ -0,0 +1,75 @@
+To implement the `job` interface, the newly created jobs must implement the `execute(*config.Config, kubernetes.Interface)` function, which takes the installation config and a kubernetes interface as input, so that the jobs can interact with the cluster. The return value must be an error. Additionally, the `when()` function must be implemented, which returns the component the job is bound to and whether it should run pre or post the deployment. The `identify()` function also needs to be implemented to have a unique identifier for each job. If the active solution for tagging jobs as deprecated is chosen, then the `deprecate` function also must be implemented - more in the next section.
+
+The jobManager is used by the `deployment` package and in the `engine` package . At the hooks, during the deployment phase, each hook only has to check if the key for the wanted component is present in the pre- or post-map. If it's present, the jobs in the map are trigged, if not, nothing must be done.
+
+Retries for the jobs are not handled by the jobManager. Retries should be implemented by the jobs themselves, because it's more flexible and the interface is easy to manage. Also, the check if the logic of the job should be executed stays inside of the job, and is not implemented by the jobManager.
+
+### Deprecation of Jobs
+
+At this point, we will stick to the passiv solution, we will may switch to the active solution in the future. If no deprecation version is decided or not necessary, should also be documented.
+
+Passive:
+- Tag (as a comment above the register call) at which defined Kyma version the jobs should be deprecated.
+
+Active:
+- Go-Build-Tags cannot be used for this use case.
+- Add `deprecation` function to job-interface, which returns at which Kyma version the job should be deprecated. Before the job is executed, the deprecation function is called to check whether it is already deprecated. If deprecated, an Error should be thrown to block the CI.
+
+### Workflow of the jobManager
+<img src="./pictures/migration-logic-diagram.png?raw=true">
+
+
+## Trade-Offs
+
+The jobs of this package cannot handle every situation that may come up in the cluster, because we do not know what the setup/usage of the customer's cluster looks like - for example, which provisioner is used, and especially regarding the access rights of the user  who is deploying Kyma. Thus, an additional migration guide will be needed in the future, as before. Let us demonstrate this on the `increaseLoggingPvcSize job`:
+- The option `allowVolumeExpansion` of the respective `PVC` must be set to `true`. If it's not, it must be changed. To do this, the provided kubeconfig must have admin rights, and the hypervisor must allow it:
+   - __k3d:__ Using a local cluster to deploy Kyma on the sample job fails, since k3d is missing a plugin to expand existing volumes. 
```

Comment:
```suggestion
   - __k3d:__ Using a local cluster to deploy Kyma on the sample job fails, because k3d is missing a plugin to expand existing volumes. 
```

Is k3d _missing_ it (wants to have it) or rather _lacking_ it (doesn't have it)? Or should we simply say "k3d doesn't have a plugin..."

Reply:
It is lacking this plugin, and as far as I know they do not have anything planed to add it.